kerlescan/profile_parser.py

```diff
@@ -25,8 +26,22 @@ def _parse_lists_of_strings(names, verb):
         helper method to convert lists of strings to comparable facts
         """
         for list_of_strings in names:
-            for item in system_profile.get(list_of_strings, []):
-                parsed_profile[list_of_strings + "." + item] = verb
+            if verb:
+                for item in system_profile.get(list_of_strings, []):
+                    parsed_profile[list_of_strings + "." + item] = verb
+            else:
+                if list_of_strings in parsed_profile:
+                    if not isinstance(parsed_profile[list_of_strings], list):
+                        starter_item = parsed_profile[list_of_strings]
+                        parsed_profile[list_of_strings] = [starter_item]
+                else:
+                    parsed_profile[list_of_strings] = []
+                for item in system_profile.get(list_of_strings, []):
+                    if item.startswith("gpg-pubkey-"):
```

Comment:
Observation: Hardcoded concrete case handling within a general parse function.

Why is it necessary to limit the value to 11 characters?

Reply:
so, we only want to display the alpha-numeric key portion, but when someone created these objects in the reporter, they structured them to have the string start with "gpg-pubkey-" which we need to strip out.

For example, here's how they look as they come in:
![image](https://user-images.githubusercontent.com/24516652/120358525-8ed34d80-c2d4-11eb-8419-ec0b7d7118cd.png)
and here's how we want to display them:
![image](https://user-images.githubusercontent.com/24516652/120358617-ae6a7600-c2d4-11eb-9ca0-bbfa1ea8e4b4.png)
(except after this mock was created, we decided not to nest with a second fact name, just to use the category name only, but you can see the alpha-numerical key portion is displayed alone.)

Reply:
So perhaps the solution here is what you observed below; to have a specific dedicated parse function instead so that it's clear that the reason for the parsing is only due to the nature of gpg-pubkeys as anything else would need to go through a different if/else path here.

Reply:
Either a specific function, or a specific parser/filter on the value. But if this is the only fact that requires value filtering, then hardcoding the case would be the quickest way. I would maybe move the constant up (to be close to other constants).

```
if item.startswith(GPG_KEY_PREFIX):
```

Reply:
Made the suggested changes; please take another look. Thanks! :)